src/main/java/org/opentripplanner/standalone/BugsnagReporter.java

```diff
@@ -0,0 +1,77 @@
+package org.opentripplanner.standalone;
+
+import com.bugsnag.Bugsnag;
+import com.bugsnag.Report;
+import com.bugsnag.Severity;
+import org.opentripplanner.common.MavenVersion;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Bugsnag util for reporting errors to the project defined by the Bugsnag project notifier API key.
+ *
+ * A Bugsnag project identifier key is unique to a Bugsnag project and allows errors to be saved against it. This key
+ * can be obtained by logging into Bugsnag (https://app.bugsnag.com), clicking on Projects (left side menu) and
+ * selecting the required project. Once selected, the notifier API key is presented.
+ */
+public class BugsnagReporter {
+    private static Bugsnag bugsnag;
+    private static final Logger LOG = LoggerFactory.getLogger(BugsnagReporter.class);
+
+    /**
+     * Initialize Bugsnag using the project notifier API key when the application is first loaded.
+     * @param params The parsed OTP command line parameters
+     */
+    public static void initializeBugsnagErrorReporting(CommandLineParameters params) {
+        if (params.bugsnagKey != null) {
+            bugsnag = new Bugsnag(params.bugsnagKey);
+            bugsnag.setAppVersion(MavenVersion.VERSION.getShortVersionString());
+            if (params.bugsnagReleaseStage != null) {
+                bugsnag.setReleaseStage(params.bugsnagReleaseStage);
+            }
+            if (params.bugsnagAppType != null) {
+                bugsnag.setAppType(params.bugsnagAppType);
+            }
+            LOG.info("Bugsnag reporting enabled.");
+        } else {
+            LOG.warn("Bugsnag project notifier API key not available. Bugsnag error reporting disabled.");
+        }
+    }
+
+    /**
+     * If Bugsnag has been configured, report error based on provided information.
+     */
+    public static boolean reportErrorToBugsnag(String message, Throwable throwable) {
+        return reportErrorToBugsnag(message, null, throwable);
+    }
+
+    /**
+     * If Bugsnag has been configured, report error based on provided information. Note: throwable must be non-null.
+     */
+    public static boolean reportErrorToBugsnag(String message, Object badEntity, Throwable throwable) {
+        // Log error to log output.
+        LOG.error(message, throwable);
+
+        // If bugsnag is disabled, make sure to report full error to otp-middleware logs.
```

Comment:
Remove reference to otp-middleware.

Reply:
Done in 2e19ad4ff455fd4a2475d8f235d77b9b0ac3acd5